src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1921,13 +1988,16 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                 }
             }
 
-            var resultTypeWithAnnotations = TypeSymbolWithAnnotations.Create(resultType ?? node.Type.SetUnknownNullabilityForReferenceTypes(), isNullableIfReferenceType);
-
             if ((object)resultType != null)
             {
+                // Let's pretend the result type is nullable, in order to avoid warnings reported for top level nullability by ApplyConversion.
```

Comment:
>Let's pretend the result type is nullable, in order to avoid warnings reported for top level nullability by ApplyConversion. [](start = 19, length = 124)

Would using `reportTopLevelWarnings: false` option in `ApplyConversion` achieve this?

Reply:
> Would using reportTopLevelWarnings: false option in ApplyConversion achieve this?

Perhaps, but we still need to build the target type.

---
In reply to: [236889499](https://github.com/dotnet/roslyn/pull/31264#discussion_r236889499) [](ancestors = 236889499)